metecho/api/jobs.py

```diff
@@ -316,6 +316,35 @@ def create_branches_on_github_then_create_scratch_org(
 )
 
 
+def convert_to_dev_org(scratch_org, *, task, originating_user_id=None):
+    """
+    Convert an Epic playground org into a Task Dev org
+    """
+    error = None
+    task.refresh_from_db()
+    scratch_org.refresh_from_db()
+    try:
+        _create_branches_on_github(
+            user=scratch_org.owner,
+            repo_id=task.get_repo_id(),
+            epic=task.epic,
+            task=task,
+            task_sha=scratch_org.latest_commit,
+            originating_user_id=originating_user_id,
+        )
+    except Exception as exc:
+        error = exc
+        logger.exception("Failed to convert ScratchOrg")
```

Comment:
I don't know the pros/cons here, but I do notice this is a bit different than how we're handling errors elsewhere in jobs, e.g.:

```py
        scratch_org.finalize_*(error=e, originating_user_id=originating_user_id)
        tb = traceback.format_exc()
        logger.error(tb)
        raise
```

Reply:
Hmm, good question! `logger.exception` [logs on the ERROR level with exception info](https://docs.python.org/3/library/logging.html#logging.exception), so just a shortcut for the `tb` assignment and produces the same result. I think the real difference is the final `raise`, which will propagate as an unhandled exception. IMO we don't need to re-raise if we are already logging with ERROR (which should notify us via Sentry), but perhaps there's an important reason why it should be there 🤷‍♂️

Reply:
Please keep in mind that we are not using Sentry in production. An unexpected error here seems like something that should be reported to the user.

Reply:
I'd also rather keep the pattern consistent unless/until we want to change it for other jobs too.

Reply:
> An unexpected error here seems like something that should be reported to the user.

Oh yeah, I agree. The UX is identical because the `finalize_*` method is always called so the user will get a toast with either success or error information. The only real change is the final `raise`, which I guess will just bubble all they way up in the worker

> I'd also rather keep the pattern consistent unless/until we want to change it for other jobs too.

Sounds good, the [latest commit](https://github.com/oddbird/Metecho/pull/1019/commits/40aea2b477dfd97df1b26e0ed41156e8184e7d88) uses the previous style 👍

Reply:
I think the final `raise` is essential in order for `finalize_*` to also have a chance to process the error. Not re-raising here suppresses the exception.